src/huffman.hpp

```diff
@@ -248,6 +250,14 @@ class code_table
   }
 
 public:
+  /// Code point type
+  ///
+  using code_point_type = code_point<Symbol>;
+
+  /// Symbol type
+  ///
+  using symbol_type = typename code_point_type::symbol_type;
+
   /// Constructs a `code_table` from a symbol-frequency mapping
```

Comment:
unrelated to this PR, but:
The doc for code_table says:

@pre frequency for a given symbol is positive

Will things actually break if the frequency is 0? If not, we should add a test for it and update the comment. If yes then maybe we can change the code to handle that case?

Reply:
It depends on how the 0 frequency symbols are handled. They would probably need to be dropped. If maintained in the table, they'll start receiving codes - which will mean more bits are needed for symbols that actually show up.

But if the zero frequency symbols need to be removed, I think it's better to establish as a precondition and force the caller to do so.